test/e2e/create_test.go

```diff
@@ -44,50 +44,114 @@ func testWindowsNodeCreation(t *testing.T) {
+	require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
+	if tc.CloudProvider.GetType() == config.AWSPlatformType {
+		// Replace the known private key with a randomly generated one.
+		err = tc.createPrivateKeySecret(false)
+		require.NoError(t, err, "error replacing private key secret")
+	}
+	err = tc.waitForWindowsNodes(gc.numberOfMachineNodes, true, false, false, false)
+	assert.NoError(t, err, "Windows node creation failed")
+
+	t.Run("Changing the private key ensures all Windows nodes use the new private key", func(t *testing.T) {
+		// This test cannot be run on vSphere because this random key is not part of the vSphere template image.
+		// Moreover this test is platform agnostic so is not needed to be run for every supported platform.
+		if tc.CloudProvider.GetType() != config.AWSPlatformType {
+			t.Skipf("Skipping for %s", tc.CloudProvider.GetType())
+		}
+		// Replace private key and check that new Machines are created using the new private key
+		err = tc.createPrivateKeySecret(false)
+		require.NoError(t, err, "error replacing private key secret")
+		err = tc.waitForWindowsNodes(gc.numberOfMachineNodes, true, false, false, false)
+		assert.NoError(t, err, "error waiting for Windows nodes configured with newly created private key")
 	})
+}
 
-	t.Run("Windows Machines with the Windows label are configured", func(t *testing.T) {
-		_, err := testCtx.createWindowsMachineSet(gc.numberOfNodes, true)
-		require.NoError(t, err, "failed to create Windows MachineSet")
-		// We need to cover the case where a user changes the private key secret before the WMCO has a chance to
-		// configure the Machine. In order to simulate that case we need to wait for the MachineSet to be fully
-		// provisioned and then change the key. The the correct amount of nodes being configured is proof that the
-		// mismatched Machine created with the mismatched key was deleted and replaced.
-		// Depending on timing and configuration flakes this will either cause all Machines, or all Machines after
-		// the first configured Machines to hit this scenario. This is a platform agonistic test so we run it only on
-		// AWS.
-		err = testCtx.waitForWindowsMachines(int(gc.numberOfNodes), "Provisioned")
-		require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
-		if testCtx.CloudProvider.GetType() == config.AWSPlatformType {
-			// Replace the known private key with a randomly generated one.
-			err = testCtx.createPrivateKeySecret(false)
-			require.NoError(t, err, "error replacing private key secret")
+// testBYOHConfiguration tests that the ConfigMap controller can properly configure VMs
+func (tc *testContext) testBYOHConfiguration(t *testing.T) {
+	if gc.numberOfBYOHNodes == 0 {
+		t.Skip("BYOH testing disabled")
+	}
+	_, err := tc.createWindowsMachineSet(gc.numberOfBYOHNodes, false)
+	machines, err := tc.waitForWindowsMachines(int(gc.numberOfBYOHNodes), "Provisioned", false)
```

Comment:
>The issue with that is that the successful creation of the MachineSet does not imply that all the Machines have been created,

Correct. That is what `waitForWindowsMachines()` does.

> or that the Machines originally created will be the ones that hit the Provisioned state. 

This seems to imply that the list of `Machines` should be an input to `waitForWindowsMachines()` instead of it trying to heuristically figure out the `Machines`. WDYT?